### PR TITLE
Add function to process file into MezzuriteComponent

### PR DESCRIPTION
**Pull request by kawwong**

### CODE REVIEW

File: server/src/utilities/processFile/processFile.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { readFile } from 'fs';
+import Project from 'ts-morph';
+
+import MezzuriteComponent from '../../models/MezzuriteComponent';
+import getComponentType from '../getComponentType';
+import generateComponent from '../generateComponent';
+
+function processFile (filePath: string, project: Project): Promise<MezzuriteComponent> {
```
Comment (cn894) [inline server/src/utilities/processFile/processFile.ts]: There's no null check for the `project` parameter. If it's `null` or `undefined`, it won't break until line 16, which is after spending some resources in reading the file and checking its component type.

File: server/src/utilities/processFile/processFile.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  });
+
+  it('should resolve the promise with a null value when the file path does not exist', () => {
+    const filePath = 'does not exist';
+    return processFile(filePath, project).then((component: MezzuriteComponent) => {
+      expect(component).toBeNull();
+    });
+  });
+
+  it('should resolve the promise with a null value when the filePath is null', () => {
```
Comment (cn894) [inline server/src/utilities/processFile/processFile.test.ts]: I'm not sure that I agree with a Promise resolving with `null` when provided an invalid parameter. My first instinct when I think about a Promise-returning function handling invalid parameters is to reject it, as it's not a valid way of calling the function. Thoughts?

Comment (kawwong) [reply server/src/utilities/processFile/processFile.test.ts]: Agreed. Semantically the promise should reject and leave all responsibility up to the caller. Resolved in https://github.com/Microsoft/vscode-mezzurite/pull/31/commits/792b7100303dcf4fea87143696289b1e87c7d5e4.

File: server/src/utilities/processFile/processFile.ts (hunk 1 above)
Comment (kawwong) [reply server/src/utilities/processFile/processFile.ts]: Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/31/commits/792b7100303dcf4fea87143696289b1e87c7d5e4.

**Pull request merged**